src/main/java/mate/academy/bookshop/repository/book/providers/PriceSpecificationProvider.java

```diff
@@ -1,28 +1,37 @@
 package mate.academy.bookshop.repository.book.providers;
 
 import java.math.BigDecimal;
+import mate.academy.bookshop.exception.InvalidPriceRangeException;
 import mate.academy.bookshop.model.Book;
 import mate.academy.bookshop.repository.SpecificationProvider;
 import org.springframework.data.jpa.domain.Specification;
 import org.springframework.stereotype.Component;
 
 @Component
 public class PriceSpecificationProvider implements SpecificationProvider<Book> {
+    private static final String PRICE = "price";
+    private static final int EXPECTED_LENGTH = 2;
+    private static final int MIN_PRICE_INDEX = 0;
+    private static final int MAX_PRICE_INDEX = 1;
 
     @Override
     public String getKey() {
-        return "price";
+        return PRICE;
     }
 
     @Override
     public Specification<Book> getSpecification(String[] params) {
-        if (params == null || params.length != 2) {
-            throw new IllegalArgumentException("Invalid price range parameters");
-        }
-        BigDecimal minPrice = BigDecimal.valueOf(Long.parseLong(params[0]));
-        BigDecimal maxPrice = BigDecimal.valueOf(Long.parseLong(params[1]));
+        try {
+            if (params == null || params.length != EXPECTED_LENGTH) {
+                throw new InvalidPriceRangeException("Price range should contain strict 2 params");
+            }
+            BigDecimal minPrice = BigDecimal.valueOf(Long.parseLong(params[MIN_PRICE_INDEX]));
+            BigDecimal maxPrice = BigDecimal.valueOf(Long.parseLong(params[MAX_PRICE_INDEX]));
 
-        return (root, query, criteriaBuilder)
-                -> criteriaBuilder.between(root.get("price"), minPrice, maxPrice);
+            return (root, query, criteriaBuilder)
+                    -> criteriaBuilder.between(root.get(PRICE), minPrice, maxPrice);
+        } catch (NumberFormatException e) {
+            throw new InvalidPriceRangeException("Price range should contain only numbers");
+        }
     }
 }
```

Comment:
Are you sure that this logic should be in this PR?

Reply:
I thought that its good to fix this bug in pull request connected to exceptions.